services/scheduler/src/SchedulerApp.js

```diff
@@ -1,14 +1,166 @@
+                }
+                continue;
+            }
+
+            let schedulerRecord = schedulerRecordsIndex[flowModel.id];
+            if (!schedulerRecord) {
+                schedulerRecord = schedulerRecordsIndex[flowModel.id] = {
+                    'apiVersion': 'elastic.io/v1',
+                    'kind': 'SchedulerRecord',
+                    'metadata': {
+                        'name': flowModel.id,
+                        'namespace': this._config.get('NAMESPACE'),
+                        ownerReferences: [
+                            {
+                                apiVersion: 'elastic.io/v1',
+                                kind: 'Flow',
+                                controller: true,
+                                name: flowModel.metadata.name,
+                                uid: flowModel.metadata.uid
+                            }
+                        ]
+                    },
+                    'spec': {}
+                };
+            }
+
+            this._logger.info({
+                flowId: flowModel.id,
+                dueExecution: schedulerRecord.spec.dueExecution,
+                now
+            });
+            if (!schedulerRecord.spec.dueExecution || now >= new Date(schedulerRecord.spec.dueExecution)) {
+                flowsToSchedule.push(flowModel);
+            } else {
+                this._logger.trace({
+                    flowId: flowModel.id,
+                    dueExecution: schedulerRecord.spec.dueExecution
+                }, 'skip schedule');
+            }
+        }
+
+        // for (let flowId in schedulerRecordsIndex) {
+        //     await this._crdClient.schedulerrecords(flowId).delete();
+        // }
+
+        return flowsToSchedule;
+    }
+
+    async planNextRun(flow) { //eslint-disable-line no-unused-vars
+        const schedulerRecord = this._schedulerRecordsIndex[flow.id];
+        const newDueExecution = new Date();
+        newDueExecution.setMinutes(newDueExecution.getMinutes() + TICK_INTERVAL_MINUTES);
+        this._logger.trace({flowId: flow.id, dueExecution: newDueExecution}, 'schedule next flow tick');
+        schedulerRecord.spec.dueExecution = newDueExecution;
+        if (!schedulerRecord.metadata.uid) {
+            this._logger.trace({
+                name: schedulerRecord.metadata.name,
+                dueExecution: schedulerRecord.spec.dueExecution
+            }, 'Create new scheduler record');
+            await this._crdClient.schedulerrecords.post({body: schedulerRecord});
+        } else {
+            this._logger.trace({
+                name: schedulerRecord.metadata.name,
+                dueExecution: schedulerRecord.spec.dueExecution
+            }, 'Update new scheduler record');
+            await this._crdClient.schedulerrecords(flow.id).put({body: schedulerRecord});
+        }
+    }
+}
+
+class OIH_SchedulePublisher extends SchedulePublisher {
```

Comment:
AFAIK this class should be common for scheduler and webhooks. They both do almost the same